components/ImgPreviewer.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div
+    :class="[
+      'flex',
+      'justify-center',
+      { 'w-[138px]': props.size === 'large' },
+      { 'w-[80px]': props.size === 'small' },
+      'mr-[16px]',
+      'overflow-hidden',
+    ]"
+  >
+    <img
+      v-if="computedFileType === 'image'"
+      :class="['w-full', 'h-auto', 'object-contain', 'rounded-[8px]']"
+      :src="props.fileData"
+    >
+    <UIcon
+      v-else-if="computedFileType === 'epub'"
+      name="i-heroicons-book-open"
+      class="text-dark-gray w-[40px] h-[40px]"
+    />
+    <UIcon
+      v-else
+      name="i-heroicons-document"
+      class="text-dark-gray w-[40px] h-[40px]"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useFileUpload } from '~/composables/useFileUpload'
+
+type FileTypes = 'image' | 'pdf' | 'epub' | 'other'
+
+const { getFileType } = useFileUpload()
+
+const props = defineProps({
+  fileType: {
+    type: String as PropType<FileTypes>,
+    default: undefined
+  },
+  fileData: { type: String, default: undefined },
+  size: { type: String, default: 'large' }
+})
+
+const computedFileType = computed(() => {
+  if (props.fileType) { return props.fileType }
+  return getFileType(props.fileData?.split(';')[0].split(':')[1] || '')
```

Comment:
What is the expected format of `props.fileData`? Can we clean this code in more readable format or at least add comment?

Reply:
[🎨 Make computedFileType more readable](https://github.com/likecoin/nft-book-press/pull/290/commits/1c8b9f387e476970caac4acde98465449df6a2b6)